### PR TITLE
[RLlib] PPO Tf RL Module

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: Avnish <avnishnarayan@gmail.com>

I broke the PPO Tf RL Module and its tests into a separate PR.

I also moved the loss util of the PPO Torch RL Module around and modified its tests to incorporate the TF RL Module.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/models/tf/primitives.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+from typing import List
```
Comment (kouroshHakha) [inline rllib/models/tf/primitives.py:1]: At some point we will replace these with ModelCatalog.

File: rllib/algorithms/ppo/torch/ppo_torch_rl_module.py [hunk 2]
```diff
@@ -246,7 +246,7 @@ def output_specs_exploration(self) -> SpecDict:
         specs = {SampleBatch.ACTION_DIST: self.__get_action_dist_type()}
         if self._is_discrete:
             specs[SampleBatch.ACTION_DIST_INPUTS] = {
-                "logits": TorchTensorSpec("b, h", h=self.config.action_space.n)
+                "logits": TorchTensorSpec("b, h", h=int(self.config.action_space.n))
```
Comment (kouroshHakha) [inline rllib/algorithms/ppo/torch/ppo_torch_rl_module.py:249]: curious to know why `.n` is not an int already?

**Review by kouroshHakha — APPROVED** [review]
LGTM

File: rllib/algorithms/ppo/torch/ppo_torch_rl_module.py (hunk 2 above)
Comment (avnishn) [reply rllib/algorithms/ppo/torch/ppo_torch_rl_module.py:249]: its a numpy.int64

File: rllib/algorithms/ppo/tests/test_ppo_rl_module.py [hunk 3]
```diff
@@ -105,126 +207,151 @@ def setUpClass(cls):
...
+
     def test_rollouts(self):
         # TODO: Add BreakoutNoFrameskip-v4 to cover a 3D obs space
-        for env_name in ["CartPole-v1", "Pendulum-v1"]:
-            for fwd_fn in ["forward_exploration", "forward_inference"]:
+        for fw in framework_iterator(frameworks=["torch", "tf2"]):
+            for env_name in ["CartPole-v1", "Pendulum-v1"]:
+                for fwd_fn in ["forward_exploration", "forward_inference"]:
+                    for shared_encoder in [False, True]:
+                        for lstm in [True, False]:
```
Comment (gjoliver) [inline rllib/algorithms/ppo/tests/test_ppo_rl_module.py:234]: these for loops look ... you know
can you use itertools instead?
https://stackoverflow.com/questions/798854/all-combinations-of-a-list-of-lists

**Review by gjoliver — APPROVED** [review]
pretty clean. just have 1 random suggestion.
